torch/csrc/jit/init.cpp

```diff
@@ -406,16 +406,41 @@ void initJITBindings(PyObject* module) {
 
   py::class_<PyTorchStreamWriter>(m, "PyTorchFileWriter")
       .def(py::init<std::string>())
+      .def(py::init<const std::function<size_t(const void*, size_t)>&>())
       .def(
           "write_record",
           [](PyTorchStreamWriter& self,
              const std::string& name,
              const char* data,
              size_t size) { return self.writeRecord(name, data, size); })
-      .def("write_end_of_file", &PyTorchStreamWriter::writeEndOfFile);
+      .def("write_end_of_file", &PyTorchStreamWriter::writeEndOfFile)
+      .def(
+          "write_record",
+          [](PyTorchStreamWriter& self,
+             const std::string& name,
+             uintptr_t data,
+             size_t size) {
+            return self.writeRecord(
+                name, reinterpret_cast<const char*>(data), size);
+          });
+
+  m.def("make_bytes", [](py::capsule data, size_t len) {
```

Comment:
My overall impression here is that this API is hard to follow. There are a lot of crosses between python and C++ (make_bytes, read_into)  that have to be tracked down. Our experience with the torchscript/python bindings is that this stuff grows and becomes inscrutable. The interface between Python and C++ should be as narrow as possible.

Reply:
I agree but I don't think it's avoidable here within miniz's interface since Python doesn't have built in utilities to work with pre-allocated buffers. The pathway of Python  (torch.save/load) -> C++ (miniz) -> Python (for a generic `pickle`) -> C++ (for writing to a buffer) was the only way I found that worked

Reply:
I would expect to see an API for writing a zip file, and calls from Python to use that API. It is fine if the bound methods for that API take python objects and there is some code in C++ to convert the python object to a buffer), but directly exposing things like make_bytes and swapping back and forth does not make it easy to follow.